### PR TITLE
Add SVI Profile capability

### LINKED ISSUE #349: Add support of profile at SVI level
**Issue Type**

- Role enhancement: `eos_l3ls_evpn`

**Is your feature request related to a problem? Please describe.**

When a large number has to be configured it takes a lot of time to create/update information as we have to go SVI per SVI.

**Describe the solution you'd like**

It can be easier to use a set of PROFILE to configure common parameters only one time and apply a profile on SVI.

This profile would be optional and the user can choose to configure all parameters on SVI or a profile.

Below is a logical representation:

```yaml
profile_svi:
  MYPROFILE:
    tags: [DC1]
    enabled: true
    igmp_snooping_enabled: false

tenants:
  # Tenant A Specific Information - VRFs / VLANs
  Tenant_A:
    mac_vrf_vni_base: 10000
    vrfs:
      TENANT_A_PROJECT01:
        vrf_vni: 11
        svis:
          110:
            name: 'PR01-DMZ'
            ip_address_virtual: 10.1.10.254/24
    		profile: MYPROFILE
          111:
            name: 'PR01-TRUST'
            tags: [DC1]
            enabled: true
            ip_address_virtual: 10.1.11.254/24
```

Idea from @ClausHolbechArista

### LINKED ISSUE #352: SVI State not applied to configuration
## Issue Type

- Bug Report

## Summary

SVI state management is not applied correctly. SVI are always UP even with 

```
tenants:
  # Tenant A Specific Information - VRFs / VLANs
  Tenant_A:
    mac_vrf_vni_base: 10000
    vrfs:
      TENANT_A_PROJECT01:
        vrf_vni: 11
        svis:
          110:
            name: 'PR01-DMZ'
            tags: [DC1]
            enabled: false
            ip_address_virtual: 10.1.10.254/24
```

## Role or Module Name

`eos_l3ls_evpn`

### PR DESCRIPTION
**Pull request by titom73**
## Change Summary

Implement profile management to apply on SVI interface

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #349 
Fixes #352

## Component(s) name

Role: `eos_l3ls_evpn`

## Proposed changes

Create a structure named `svi_profiles` to define a list of pre-configured parameters to apply on multiple SVI.

A new key is created at SVI level to apply profile.

Once PR #337 and #345 will be merged, these features will be applied as well.

## How to test

In Tenant Networks configuration, create `svi_profiles` structure:

```yaml
---
# SVI Profile
svi_profiles:
  GENERIC_DC_PROFILE:
    mtu: 1500
    enabled: false
    igmp_snooping: false
    ip_helpers:
      1.1.1.1:
        source_interface: lo100
        source_vrf: TENANT_A_PROJECT01
```

In SVI configuration, apply profile when required:

```yaml
tenants:
  # Tenant A Specific Information - VRFs / VLANs
  Tenant_A:
    mac_vrf_vni_base: 10000
    vrfs:
      TENANT_A_PROJECT01:
        vrf_vni: 11
        svis:
          110:
            name: 'PR01-DMZ'
            tags: [DC1]
            enabled: true
            ip_address_virtual: 10.1.10.254/24
            igmp_snooping_enabled: false
          112:
            tags: [DC1]
            name: 'PR01-DMZ'
            profile: GENERIC_DC_PROFILE
            ip_address_virtual: 10.1.11.254/24
```

## Todo

- [x] Update documentation
- [ ] Implement keys related to #337 and #345 

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

**Comment by carlbuchmann** [comment]
Requires #345 to be merged and incorporated before proceeding.

**Comment by blemmenes** [comment]
Is this expected to get backported in to v1.1.x or will we have to cherry-pick this to get the fix for #352 ?

**Comment by carlbuchmann** [comment]
> Is this expected to get backported in to v1.1.x or will we have to cherry-pick this to get the fix for #352 ?

@blemmenes - I just updated the milestone, this will be targeted for release 1.1.1 - https://github.com/aristanetworks/ansible-avd/milestone/6

**Comment by titom73** [comment]
MKdocs issue seems to be a DNS resolution issue. Can be ignored since, after manual check, the link is working

**Comment by titom73** [comment]
Ready for review

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/tenant_evpn_vxlan/leaf-tenant-vlan-interfaces.j2 [hunk 1]
```diff
@@ -21,19 +38,36 @@
 {%                          endif %}
 {%                     endfor %}
 {%                 endif %}
-{%                 if tenants[tenant].vrfs[vrf].svis[svi].ip_virtual_router_address is defined %}
-    ip_virtual_router_address: {{ tenants[tenant].vrfs[vrf].svis[svi].ip_virtual_router_address | ipaddr('address') }}
+{# Virtual Router IP Address #}
+{%                 if profile.per_svi.ip_virtual_router_address is defined or profile.svi_profile.ip_virtual_router_address is defined %}
+    ip_virtual_router_address: {{ profile.per_svi.ip_virtual_router_address if profile.per_svi.ip_virtual_router_address is defined else profile.svi_profile.ip_virtual_router_address }}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/tenant_evpn_vxlan/leaf-tenant-vlan-interfaces.j2]: ip_virtual_router_address needs to render only the address and not the mask
update logice to include ipaddr filter: ```{{ profile.per_svi.ip_virtual_router_address | ipaddr('address') if profile.per_svi.ip_virtual_router_address is defined else profile.svi_profile.ip_virtual_router_address | ipaddr('address') }}```

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/tenant_evpn_vxlan/leaf-tenant-vlan-interfaces.j2 [hunk 2]
```diff
@@ -6,12 +6,29 @@
 ## {{ tenant }} ##
 {%             endif %}
 {%             for svi in tenants[tenant].vrfs[vrf].svis | arista.avd.natural_sort if svi in leaf.svis %}
+{# Detect if a svi_profile exists #}
+{# If exists, create a shortpath to access profile data #}
+{%                 set profile = namespace() %}
+{%                 set profile.exist = false %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/tenant_evpn_vxlan/leaf-tenant-vlan-interfaces.j2]: `{% set profile.exist = false %}` doesn't appear to be used elsewhere in the code, would suggest removing!

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/tenant_evpn_vxlan/leaf-tenant-vlan-interfaces.j2 [hunk 3]
```diff
@@ -6,12 +6,29 @@
...
+{%                 set profile.per_svi = tenants[tenant].vrfs[vrf].svis[svi] %}
+{%                 if svi_profiles is defined and svi_profiles is not none %}
+{%                     if tenants[tenant].vrfs[vrf].svis[svi].profile is defined %}
+{%                         if tenants[tenant].vrfs[vrf].svis[svi].profile in svi_profiles %}
+{%                             set profile.exist = true %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/tenant_evpn_vxlan/leaf-tenant-vlan-interfaces.j2]: {% set profile.exist = true %} doesn't appear to be used elsewhere in the code, would suggest removing!

**Review by carlbuchmann — CHANGES_REQUESTED** [review]
Some slight corrections before this can be merged, please in-line comments!

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/README.md [hunk 4]
```diff
@@ -811,6 +811,20 @@ evpn_rd_type:
...
   admin_subfield: < "leaf_asn" | "spine_asn" | "vni" | <0-65535> | <0-4294967295> | default -> "vni" >
 
+# Optional profiles to apply on SVI interfaces
+# Each profile can support all or some of the following keys according your own needs.
+# Keys are the same used under SVI.
+svi_profiles:
+  < profile_name >:
+    mtu: < mtu >
+    enabled: < true | false >
+    ip_address_virtual: < IPv4_address/Mask >
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/README.md:822]: add:
```yaml
ip_virtual_router_address: < IPv4_address/Mask >
ip_address_virtual_secondary: < IPv4_address/Mask >
```

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/tenant_evpn_vxlan/leaf-tenant-igmp-snooping.j2 [hunk 5]
```diff
@@ -34,10 +36,25 @@
 {%              for vrf in tenants[tenant].vrfs | arista.avd.natural_sort if leaf.vrfs is not none and vrf in leaf.vrfs %}
 {# Tenant VLANs w/SVIs #}
 {%                  for svi in tenants[tenant].vrfs[vrf].svis | arista.avd.natural_sort if leaf.svis is not none and svi in leaf.svis %}
-{%                      if tenants[tenant].vrfs[vrf].svis[svi].igmp_snooping_enabled is defined %}
-{%                          if tenants[tenant].vrfs[vrf].svis[svi].igmp_snooping_enabled == false and leaf.igmp_snooping_enabled == true %}
+{# Detect if a svi_profile exists #}
+{# If exists, create a shortpath to access profile data #}
+{%                      set profile = namespace() %}
+{%                      set profile.exist = false %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/tenant_evpn_vxlan/leaf-tenant-igmp-snooping.j2]: Update with new namespace format
`per_svi.values` & `profiles.values`

**Review by carlbuchmann — APPROVED** [review]
LGTM - please validate my updates!

**Pull request merged**